### DISCUSSION TITLE
How to connect to host with IPv6 address with libcurl

### DISCUSSION THREAD

**Discussion by ganeshkamath89** [curl/curl#6651] [Q&A] [answered]
I was using IPv4 address all these days to connect from my curl client to remote server application.
For some reason when I try my IPv6 Link Local address in place of IPv4 address connection is failing.

Are there any curl option header headers to enable IPv6 connection?
**Eg:**
```
IPv4 Address: 192.168.1.4
IPv6 Address: fe80::7dc9:8479:5da5:f561%18
```

**Ping is successful:**
```
ping fe80::7dc9:8479:5da5:f561%18

Pinging fe80::7dc9:8479:5da5:f561%18 with 32 bytes of data:
Reply from fe80::7dc9:8479:5da5:f561%18: time<1ms
Reply from fe80::7dc9:8479:5da5:f561%18: time<1ms
Reply from fe80::7dc9:8479:5da5:f561%18: time<1ms
Reply from fe80::7dc9:8479:5da5:f561%18: time<1ms
```

**Environment:**
OS: Windows 10 Enterprise
libCurl: 7.74
Building client from Curl source

**Comment by jzakrzewski** [comment, marked as answer]
If memory serves, you should surround the address with square brackets like:
`curl -g -6 '[fe80::7dc9:8479:5da5:f561%18]'`

**Reply by ganeshkamath89** [reply]
Hi @jzakrzewski. Is there any alternative to adding the square brackets manually?
Because in IPv6 network for hostname we are not making any changes, it looks like only for the IPv6 address format this needs to be done.

Why do we need to add the square brackets?

**Reply by bagder** [reply]
because curl works with URLs and that's how you specify a IPv6 address in a URL - it just also happens that curl supports URLs without a leading `scheme://` but a numerical address still needs to be provided "properly"

**Reply by ganeshkamath89** [reply]
Hi @bagder. Thank you for your response.
Does libcurl itself provide any mechanism to check if the hostname is a valid ipv6 format?
I am asking so that I can use libcurl APIs while adding the Square Braces

**Reply by bagder** [reply]
Sure, libcurl has a URL API that you can use to verify that a URL is okay by curl. See [curl_url_set](https://curl.se/libcurl/c/curl_url_set.html) as a starting-point.

**Reply by ganeshkamath89** [reply]
Hi @bagder. When I try with my IP Address: **fe80::7dc9:8479:5da5:f561%18**
I am getting an error: 
`URL malformat`

This is the first time I am adding IPv6 filter in Wireshark. I tried, 
`ipv6.addr==fe80::7dc9:8479:5da5:f561%18`

But the display filter is not accepting it. Do you have any idea what I am doing wrong?

**Reply by bagder** [reply]
Yes, it (the libcurl URL API) requires a URL and thus it wants brackets around the IPv6 address...

**Reply by ganeshkamath89** [reply]
Hi @bagder, I have got my application working. thank you for all the useful tips you have shared.
To filter values in wireshark, i used the filter IPv6.
